### DISCUSSION TITLE
How to use it with Prometheus?

### DISCUSSION THREAD

**Discussion by denisidoro** [marcusolsson/grafana-treemap-panel#18] [Q&A] [answered]
<img width="337" alt="Screen Shot 2021-05-21 at 07 25 07" src="https://user-images.githubusercontent.com/3226564/119123226-b9382780-ba05-11eb-9993-2c2e2c667816.png">

<img width="305" alt="Screen Shot 2021-05-21 at 07 26 30" src="https://user-images.githubusercontent.com/3226564/119123411-e7b60280-ba05-11eb-8ed6-111174d97087.png">

<img width="507" alt="Screen Shot 2021-05-21 at 07 27 24" src="https://user-images.githubusercontent.com/3226564/119123519-05836780-ba06-11eb-97f0-fdbb4203eefb.png">

`fin6_gross` in the example above has an `id` field, which could be used by `label by`:

<img width="1042" alt="Screen Shot 2021-05-21 at 07 30 03" src="https://user-images.githubusercontent.com/3226564/119123867-690d9500-ba06-11eb-9ca4-a88013c14318.png">

**Comment by marcusolsson** [comment, marked as answer]
The field used for the Text dimension needs to be a string field. I suspect that it's a Number field perhaps?

**Comment by Lercher** [comment]
I consider this a valid question, too. Especially as I'm personally not at all familiar with Grafana's inner working on Prometheus data sources. It's very reasonable that a treemap requests a series of key value pairs, and having e.g. a relational data source like SQL it's probably not that hard to provide such a result set.

But Prometheus labeled time series queries seem to be different here. Looking at the last Graph Visualization in the last image, there is a string labeled bunch of time based values with label `id`. I guess the intention is to use the label `id` as the text field and its last numeric value as the number field. See the image below for a similar query inspector result for such a quite typical PromQL query like `sum(temporal_temporal_request) by (operation)` returning N time series organized by a single label. In my example it's a time-based evolution of a count of requests grouped by the request's operation name.

So, my question is, too: How do we need to shape or layout Prometheus based queries returning plain labeled time series in Grafana to be visualized by this component? Referencing the image below e.g., in this case I'd expect the component to display a rectangular area of value 12 labeled as "RespondWorkflowTaskCompleted" as well as 3 other rectangles labeled by `operation` with values not shown for brevity. How would I need to enter one or multiple queries to get such a viz?

Thank you very much, Martin.

<img width="389" alt="image" src="https://user-images.githubusercontent.com/5909334/146643489-867cd5dc-977e-423b-8eb6-7ac6b1e65a3f.png">

**Comment by m1keil** [comment]
For anyone who's having difficulties with regards of this visualization and Prometheus metric:

- Assuming you have a query in the form of `sum by (label) (metric{..})`,
- Add "Labels to fields" transformation in "Columns" mode.

**Comment by fvigneault** [comment]
For me, the solution was to set the grafana prometheus query to Type=Instant and Format=Table. Then I was able to select the Label By and Size By in the panel settings.
